### PR TITLE
feat(ui-shell): add platform nav

### PR DESCRIPTION
**Pull request by joshblack**
Closes https://ibm.biz/BdYG4v

Adds the Platform Navigation component.

#### Changelog

**New**

- Platform Navigation
  - `_platform-nav.scss`
  - `platform-nav.hbs`
  - `ui-shell.config.js` updated to include nav options

**Changed**

**Removed**

#### Testing / Reviewing

Staging Links:

- [Category collapsed](http://carbon-dev-environment-nonlacteous-lich.mybluemix.net/component/ui-shell--platform-nav-expanded)
- [Category expanded](http://carbon-dev-environment-nonlacteous-lich.mybluemix.net/component/ui-shell--platform-nav-category-expanded)

### CODE REVIEW

File: src/components/ui-shell/_platform-nav.scss [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    font-weight: 600;
+  }
+
+  .#{$prefix}--platform__category--expanded .#{$prefix}--platform__category-title > svg {
+    transform: rotate(180deg);
+  }
+
+  .#{$prefix}--platform__category--expanded .#{$prefix}--platform__category-items {
+    display: block;
+    visibility: visible;
```
Comment (asudoh) [inline src/components/ui-shell/_platform-nav.scss]: [`visibility:visible` is a tricky thing](https://jakearchibald.com/2014/visible-undoes-hidden/) - `visibility:inherit` yields more intuitive result.

File: src/components/ui-shell/_platform-nav.scss [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+@mixin platform-nav {
+  //----------------------------------------------------------------------------
+  // Platform Navigation
+  //----------------------------------------------------------------------------
+  .#{$prefix}--platform__nav {
+    position: absolute;
+    top: get-fixed-size(6);
+    bottom: 0;
+    left: 0;
+    background-color: $ibm-colors__gray--90;
```
Comment (asudoh) [inline src/components/ui-shell/_platform-nav.scss:16]: Need theme tokens for `$ibm-colors__` refs in this file.

**Review by asudoh — COMMENTED** [review]
Great to see the progress - A couple of comments. Thanks @joshblack!

File: src/components/ui-shell/_platform-nav.scss (hunk 2 above)
Comment (joshblack) [reply src/components/ui-shell/_platform-nav.scss:16]: We don't have any for this one yet, sadly :/

Comment (asudoh) [reply src/components/ui-shell/_platform-nav.scss:16]: 😢

**Review by asudoh — APPROVED** [review]
Thank you for the update @joshblack!

**Review by lovemecomputer — APPROVED** [review]

**Review by aagonzales — APPROVED** [review]
Looks good! 👍 ✅

**Pull request merged**